src/line_launcher/mod.rs

```diff
@@ -1,66 +1,24 @@
 use midir::MidiOutputConnection;
 use rand::Rng;
-use std::sync::mpsc::Receiver;
-use wmidi::{Channel, MidiMessage, Note, Velocity};
+use std::sync::{
+    mpsc::{Receiver, Sender},
+    Arc, Mutex,
+};
+use std::time::{Duration, SystemTime};
 
-use crate::{BeatNumber, Chord, Line, Progression};
+use crate::{BeatNumber, Line, Progression};
 
-#[derive(Clone, Copy, Debug)]
-enum PlayingState {
-    NotPlaying,
-    Playing {
-        line_index: usize,
-        next_note_index: usize,
-        pitch_offset: i8,
-    },
-}
+mod midi_message_sender;
+use midi_message_sender::MidiMessageSender;
 
-enum ProgressionChordIndexState {
-    HaventStarted,
-    AtChordIndex(usize),
-}
+mod note_off_triggerer;
+use note_off_triggerer::{NoteOffInstruction, NoteOffTriggerer};
 
-struct ProgressionState<'progression> {
-    progression: &'progression Progression,
-    chord_index_state: ProgressionChordIndexState,
-}
+mod playing_state;
+use playing_state::PlayingState;
```

Comment:
Is this a somewhat idiomatic way to bring these into scope from the extracted modules?

Also I was surprised that just doing this enabled me to then reference eg `super::PlayingState` from one of the extracted modules (again is that idiomatic?)

Reply:
Yep, that's what I would do. I tend to only import using `super` for things that aren't available via a regular crate import (because they are private to the module), but I don't really know what people consider idiomatic.